var/spack/repos/spack_repo/builtin/packages/variorum/package.py

```diff
@@ -67,31 +75,59 @@ def cmake_args(self):
 
         cmake_args.append("-DJANSSON_DIR={0}".format(spec["jansson"].prefix))
 
+        # The examples remain in the build directory
+        cmake_args.append("-DBUILD_EXAMPLES=ON")
+        cmake_args.append("-DBUILD_TESTS=ON")
+
         if spec.satisfies("%cce"):
             cmake_args.append("-DCMAKE_C_FLAGS=-fcommon")
             cmake_args.append("-DCMAKE_CCC_FLAGS=-fcommon")
             cmake_args.append("-DCMAKE_Fortran_FLAGS=-ef")
 
-        if "+shared" in spec:
-            cmake_args.append("-DBUILD_SHARED_LIBS=ON")
-        else:
-            cmake_args.append("-DBUILD_SHARED_LIBS=OFF")
-
-        if "+docs" in spec:
-            cmake_args.append("-DBUILD_DOCS=ON")
-            sphinx_build_exe = join_path(spec["py-sphinx"].prefix.bin, "sphinx-build")
-            cmake_args.append("-DSPHINX_EXECUTABLE=" + sphinx_build_exe)
+        # CPU architecture selection.
+        if any( s in spec for s in ("cpu_Intel", "cpu=INTEL", "cpu=intel") ):
```

Comment:
Why not enforce a single value here? `cpu=intel`? Why allow user to specify the same build configuration using three different cases: Intel, INTEL, and intel? We wouldn't normally do this. They all describe the same actual build configuration, right?

Reply:
@eugeneswalker I did that because I kept having to look up with capitalization to use, and if that's an annoyance for me, it's probably an annoyance for the users as well.  No objection to standardizing on just one if that's the house style, though.

Reply:
Ok, I see. Yes, let us standardize the case to all lowercase as that is what is normally done. The other thing, once there is a single case used, the idiomatic way of checking if the variant is set is:
```
if spec.satisfies("cpu=ibm"):
  ...
elif spec.satisfies("cpu=intel"):
  ...
...
```

Thanks for making these updates! This is really great.